### PR TITLE
Make functions use tempname to create new files

### PR DESCRIPTION
**Pull request by genuinelucifer**
Fixing #394 
Had to change how the file is written in order to properly load a files from ```/tmp/```

- [x] Make fcode use tempname

### CODE REVIEW

**Comment by genuinelucifer** [comment]
@cbm755 Please review. I'm looking into other files as well. Will add them soon.

**Comment by genuinelucifer** [comment]
I've gone through all the files and no other file is creating any new file. So, I guess this PR is complete.
.
Please let me know if I missed something.

File: inst/@sym/function_handle.m [hunk 1]
```diff
@@ -115,7 +115,10 @@
       param.fname = param.fname(1:end-2);
     end
 
-    fname2 = param.fname; fcnname = param.fname;
+    [fcnpath, fcnname, fcnext] = fileparts(param.fname);
+    fcnname (fcnname == '-') = '_';
```
Comment (cbm755) [inline inst/@sym/function_handle.m]: let's not do this here: if user wants `-` let her do so...  (We'll fix the tempname thing elsewhere...)

File: inst/@sym/function_handle.m [hunk 2]
```diff
@@ -246,23 +253,27 @@
 %!test
 %! % output to disk
 %! fprintf('\n')
-%! f = function_handle(2*x*y, 2^x, 'vars', {x y z}, 'file', 'temp_test_output1');
+%! temp_file = tempname();
+%! temp_file(temp_file =='-') = '_';
```
Comment (cbm755) [inline inst/@sym/function_handle.m]: `temp_file = tempname('', 'oct_')`

I'm not sure its guaranteed to have to `-` but at least it doesn't by default ;-)

File: inst/@sym/function_handle.m [hunk 3]
```diff
@@ -130,11 +133,15 @@
...
+    [fid,msg] = fopen(file_to_write, 'w');
     assert(fid > -1, msg)
     fprintf(fid, '%s', M.code)
     fclose(fid);
-    fprintf('Wrote file %s.\n', M.name);
+    fprintf('Wrote file %s.\n', file_to_write);
+    if (exist(fcnpath))
+        addpath(fcnpath);
+    end
     f = str2func(fcnname);
```
Comment (cbm755) [inline inst/@sym/function_handle.m]: this is scary: I sort of think its the user's responsbility, if she wants.  Plus added `/tmp` to the path is bound to be some sort of code-injection security bug (Bob create `/tmp/ls.m` which contains bad code for Alice to run etc).

**Comment by genuinelucifer** [comment]
@cbm755 I did the changes. Please review again.

**Comment by cbm755** [comment]
How do we know `tempname` will be in `/tmp`?

**Comment by genuinelucifer** [comment]
> How do we know tempname will be in
> /tmp?

Well, i always found them there :P
.
I changed it now. If its okay, i could squash the commits.

**Comment by cbm755** [comment]
I think this is ok as-is.  I'll merge it

**Pull request merged**